wrangler-core/src/main/java/io/cdap/directives/column/SetType.java

```diff
@@ -22,44 +22,81 @@
 import io.cdap.wrangler.api.Arguments;
 import io.cdap.wrangler.api.Directive;
 import io.cdap.wrangler.api.DirectiveExecutionException;
+import io.cdap.wrangler.api.DirectiveParseException;
 import io.cdap.wrangler.api.ExecutorContext;
+import io.cdap.wrangler.api.Optional;
 import io.cdap.wrangler.api.Row;
 import io.cdap.wrangler.api.annotations.Categories;
 import io.cdap.wrangler.api.lineage.Lineage;
 import io.cdap.wrangler.api.lineage.Mutation;
+import io.cdap.wrangler.api.parser.Bool;
 import io.cdap.wrangler.api.parser.ColumnName;
 import io.cdap.wrangler.api.parser.Identifier;
+import io.cdap.wrangler.api.parser.Numeric;
+import io.cdap.wrangler.api.parser.Text;
 import io.cdap.wrangler.api.parser.TokenType;
 import io.cdap.wrangler.api.parser.UsageDefinition;
 import io.cdap.wrangler.utils.ColumnConverter;
 
+import java.math.RoundingMode;
 import java.util.List;
 
 /**
- * A Wrangler step for converting data type of a column
+ * A Wrangler step for converting data type of column
  * Accepted types are: int, short, long, double, float, string, boolean and bytes
+ * When decimal type is selected, can also specify the scale and whether to round or not
  */
 @Plugin(type = "directives")
 @Name(SetType.NAME)
 @Categories(categories = {"column"})
-@Description("Converting data type of a column.")
+@Description("Converting data type of a column. Optional arguments scale, " +
+  "round and rounding-mode are used only when type is decimal.")
 public final class SetType implements Directive, Lineage {
   public static final String NAME = "set-type";
+
+  // Default scale that is assumed in CDAP output schema
+  public static final int DEFAULT_SCALE = 3;
+  public static final boolean DEFAULT_ROUND = false;
+
   private String col;
   private String type;
+  private int scale;
+  private boolean round;
+  private RoundingMode roundingMode;
+
 
   @Override
   public UsageDefinition define() {
     UsageDefinition.Builder builder = UsageDefinition.builder(NAME);
     builder.define("column", TokenType.COLUMN_NAME);
     builder.define("type", TokenType.IDENTIFIER);
+    builder.define("scale", TokenType.NUMERIC, Optional.TRUE);
+    builder.define("round", TokenType.BOOLEAN, Optional.TRUE);
+    builder.define("rounding-mode", TokenType.TEXT, Optional.TRUE);
```

Comment:
Why do we need both arguments for rounding? It looks more complex than having 1. 
Also, we should allow to set precision as well, should not we?

Reply:
1. Two arguments for rounding because if the user isn't familiar with Java rounding modes, they can just enable rounding and we can assume HALF_UP. 
2. We can't set a 'max' precision on a Java BigDecimal object. Since Wrangler infers datatypes for the output schema from the actual row values, we can't propagate the set precision to the output schema.

Reply:
I don't think introducing two parameter makes it simpler. Any assumptions is source of confusion. And it's not java-specific, you can find more or less same set everywhere, starting from https://en.wikipedia.org/wiki/Rounding

Reply:
Done, removed the 'round' argument